### ISSUE TITLE
No save button on 2.9.0-beta.1

### ISSUE THREAD

**Issue by fralla2** [pluginsGLPI/formcreator#1527]
Updated the plugins yesterday following bug correction Subtree depth on v2.9.0-beta.1 #1521 but now there's no save button when trying to create a new form or trying to modify general section of an already created form.

![No save](https://user-images.githubusercontent.com/38400135/66572037-446cee00-eb3e-11e9-9de7-882faf7a2fac.png)

Thanks
Francois

**Comment by btry** [comment]
Hi

Which version of GLPI ?

Did you checked the error log of glpi and your HTTP server ?

By the way I found today that a (very) important method was accidentally dropped. Strange that noone reported it already. You may want to test the branch support/2.9.0 with the fix. I plan to release a beta.2 quickly to avoid useless bug reports.

**Comment by fralla2** [comment]
Glpi 9.4.4

I will see the logs just after my meeting ends and will report back to you. I will also have a try on the test branch you're mentionned in about an hour or so.

Thanks
Francois

**Comment by btry** [comment]
I think I found your problem. You very likely downloaded the last revision from the archive generated by Githib, our cloned with git.

Go to the folder of formcreator and run the following command

php vendor/bin/robo build:fa-data

Starting from 2.9.0, some PHP files needs to be generated from a dependency of GLPI. If those files are missing, then the plugin will break with a fatal error.

I would recommend you use git to update your copy to not repeat the above command on each update of the code. If not, then keep in a safe place the php files in formcreator/data, and restore them after updating the plugin.

**Comment by fralla2** [comment]
I tried your command but get the following result : 
![Bug002_1](https://user-images.githubusercontent.com/38400135/66577845-096fb800-eb48-11e9-8cb0-03ac2ef6d391.png)


I have indeed downloaded the last revision of the git archive yesterday...

Note : When loading the page "https://glpi06.cscv.qc.ca/glpi/plugins/formcreator/front/form.form.php" to create a new ticket, I get the following php error in the logs: 

`[2019-10-10 10:17:42] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi-9.4.4/inc/toolbox.class.php line 658
  *** PHP Warning(2): require_once(/var/www/html/glpi-9.4.4/plugins/formcreator/inc/../data/font-awesome_9.4.php): failed to open stream: No such file or directory
  Backtrace :
  plugins/formcreator/inc/common.class.php:184
  plugins/formcreator/inc/common.class.php:184       require_once()
  plugins/formcreator/inc/common.class.php:217       PluginFormcreatorCommon::getFontAwesomePictoNames()
  plugins/formcreator/inc/form.class.php:398         PluginFormcreatorCommon::showFontAwesomeDropdown()
  inc/commonglpi.class.php:475                       PluginFormcreatorForm->showForm()
  ajax/common.tabs.php:92                            CommonGLPI::displayStandardTab()
  {"user":"141@glpi06"}`

**Comment by btry** [comment]
Yes : this is because you don't have the vendor directory populated. You need to install composer (https://getcomposer.org (latest) or your distribution's repository (may be old))

Once done, run ```composer install```, then ```vendor/bin/robo buidl:fa-data```

**Comment by fralla2** [comment]
Now I get this when executing the command : 
`[root@glpi06 formcreator]# vendor/bin/robo buidl:fa-data
  There are no commands defined in the "buidl" namespace.
`

P.S. Thanks a lot for the time you take on helping me with that problem :)

**Comment by fralla2** [comment]
``[root@glpi06 formcreator]# ls -la
total 464
drwxrwx---. 18 apache apache  4096 Oct 10 10:38 .
drwxrwx---. 12 apache apache  4096 Oct  9 09:26 ..
drwxrwx---.  2 apache apache  4096 Oct  9 09:02 ajax
-rwxrwx---.  1 apache apache  1907 Oct  9 09:02 .atoum.php
-rwxrwx---.  1 apache apache 64329 Oct  9 09:02 CHANGELOG.md
-rwxrwx---.  1 apache apache  1084 Oct  9 09:02 composer.json
-rwxrwx---.  1 apache apache 81959 Oct  9 09:02 composer.lock
drwxrwx---.  2 apache apache  4096 Oct  9 09:02 css
drwxrwx---.  2 apache apache  4096 Oct  9 09:02 data
drwxrwx---.  2 apache apache  4096 Oct  9 09:02 front
drwxrwx---.  3 apache apache  4096 Oct  9 09:02 .github
-rwxrwx---.  1 apache apache    75 Oct  9 09:02 .gitignore
-rwxrwx---.  1 apache apache 15381 Oct  9 09:02 hook.php
-rwxrwx---.  1 apache apache   509 Oct  9 09:02 icon.png
drwxrwx---.  4 apache apache  4096 Oct  9 09:02 inc
-rwxrwx---.  1 apache apache  1409 Oct  9 09:02 index.php
drwxrwx---.  3 apache apache  4096 Oct  9 09:02 install
drwxrwx---.  2 apache apache  4096 Oct  9 09:02 js
drwxrwx---.  3 apache apache  4096 Oct  9 09:02 lib
-rwxrwx---.  1 apache apache 34892 Oct  9 09:02 LICENSE.md
drwxrwx---.  2 apache apache  4096 Oct  9 09:02 locales
-rwxrwx---.  1 apache apache   314 Oct  9 09:02 package.json
drwxrwx---.  3 apache apache  4096 Oct  9 09:02 pics
-rwxrwx---.  1 apache apache 12428 Oct  9 09:02 plugin.xml
-rwxrwx---.  1 apache apache  5285 Oct  9 09:02 README.md
-rwxrwx---.  1 apache apache 22700 Oct  9 09:02 RoboFile.php
-rwxrwx---.  1 apache apache  5898 Oct  9 09:02 RoboFilePlugin.php
-rwxrwx---.  1 apache apache 38033 Oct  9 09:02 screenshot_2.png
-rwxrwx---.  1 apache apache 41246 Oct  9 09:02 screenshot.png
drwxrwx---.  2 apache apache  4096 Oct  9 09:02 scripts
-rwxrwx---.  1 apache apache 15199 Oct  9 09:02 setup.php
drwxrwx---. 10 apache apache  4096 Oct  9 09:02 tests
drwxrwx---.  2 apache apache  4096 Oct  9 09:02 tools
-rwxrwx---.  1 apache apache  2564 Oct  9 09:02 .travis.yml
drwxrwx---.  2 apache apache  4096 Oct  9 09:02 .tx
drwxr-xr-x. 22 root   root    4096 Oct 10 10:38 vendor
